### ISSUE TITLE
support --stdin-filename when passing files via stdin

### ISSUE THREAD

**Issue by vapier** [koalaman/shellcheck#2735]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
[[ $# -eq 0 ]]
```

#### Here's what shellcheck currently says:

```
$ shellcheck test.sh 

In test.sh line 2:
[[ $# -eq 0 ]]
^------------^ SC3010 (warning): In POSIX sh, [[ ]] is undefined.

For more information:
  https://www.shellcheck.net/wiki/SC3010 -- In POSIX sh, [[ ]] is undefined.
```

```
$ shellcheck - <test.sh

In - line 2:
[[ $# -eq 0 ]]
^------------^ SC3010 (warning): In POSIX sh, [[ ]] is undefined.

For more information:
  https://www.shellcheck.net/wiki/SC3010 -- In POSIX sh, [[ ]] is undefined.
```

#### Here's what I wanted or expected to see:

The output of `-` in the diagnostics is confusing, especially when passing other files.  It's also problematic when using `--external-sources` as `SCRIPTDIR` isn't set to anything useful.

Would like to have an option like `--stdin-filename` so that I can pass `test.sh` and it'll use that name when processing stdin.

**Comment by Ordoviz** [comment]
This would be useful when hacking together a simple shellcheck wrapper where parsing shellcheck's JSON output is not worth it. For example, in [markdown-shellcheck](https://github.com/Ordoviz/markdown-shellcheck) I have to use regexes to replace the `-` with the filename.

**Comment by fredrikekre** [comment]
This would also be helpful to resolve `SCRIPTDIR` directives when using stdin.

**Comment by e-kwsm** [comment]
Why not use xargs?

**Comment by vapier** [comment]
> Why not use xargs?

That seems to miss the point of the request.  You're assuming the file is available on disk in the first place.  If that were the case, you might as well run shellcheck directly on it instead of passing through xargs.
